core/metrics-core-service/src/test/java/org/hawkular/metrics/core/service/MetricsServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.assertj.core.util.DoubleComparator;
+import org.hawkular.metrics.model.AvailabilityType;
+import org.hawkular.metrics.model.DataPoint;
+import org.hawkular.metrics.model.RatioMap;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import rx.Observable;
+
+/**
+ * @author Joel Takvorian
+ */
+public class MetricsServiceImplTest {
+
+    private final MetricsServiceImpl service = new MetricsServiceImpl();
+    private final DoubleComparator dc = new DoubleComparator(0.001);
+
+    @Test
+    public void shouldBuildRatioMapSeries() throws Exception {
+        List<DataPoint<AvailabilityType>> serie1 = ImmutableList.of(
+                new DataPoint<>(0L, AvailabilityType.UP),
+                new DataPoint<>(5L, AvailabilityType.DOWN),
+                new DataPoint<>(10L, AvailabilityType.UP));
+        List<DataPoint<AvailabilityType>> serie2 = ImmutableList.of(
+                new DataPoint<>(0L, AvailabilityType.UP),
+                new DataPoint<>(7L, AvailabilityType.DOWN),
+                new DataPoint<>(10L, AvailabilityType.ADMIN));
+        List<DataPoint<AvailabilityType>> serie3 = ImmutableList.of(
+                new DataPoint<>(1L, AvailabilityType.UP),
+                new DataPoint<>(7L, AvailabilityType.DOWN));
+        Observable<Observable<DataPoint<AvailabilityType>>> observable = Observable.just(
+                Observable.from(serie1),
+                Observable.from(serie2),
+                Observable.from(serie3));
+        List<DataPoint<RatioMap>> result = service.buildRatioMapSeries(observable, Order.ASC, AvailabilityType::getText)
```

Comment:
It must not be merged/concat or flatmaped because the "buildRatioMapSeries" needs to work on each series individually then it proceed flat-mapping and reduction. The "Observable<Observable<DataPoint<AvailabilityType>>> " is the result of loading data points on several series as I expect. The unit test has as little as possible transformations to perform.